### PR TITLE
LibJS: More Temporal spec changes

### PR DESCRIPTION
**Pull request by linusg**
This includes all relevant Temporal spec changes merged today with the exception of https://github.com/tc39/proposal-temporal/commit/1f3fba8fd25ab775629d6b1161ae38e405ea3d75, which I'll do separately.

~~I expect to add more commits to this PR soon, just waiting for them to merge more approved changes:~~

image

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainMonthDay.cpp [hunk 1]
```diff
@@ -187,8 +187,8 @@ ThrowCompletionOr<String> temporal_month_day_to_string(GlobalObject& global_obje
     // 6. Let calendarID be ? ToString(monthDay.[[Calendar]]).
     auto calendar_id = TRY(Value(&month_day.calendar()).to_string(global_object));
 
-    // 7. If calendarID is not "iso8601", then
-    if (calendar_id != "iso8601"sv) {
+    // 7. If showCalendar is "always" or if calendarID is not "iso8601", then
+    if (show_calendar == "always" || calendar_id != "iso8601"sv) {
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/PlainMonthDay.cpp:191]: `"always"sv` for some SV-SV comparison

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp [hunk 2]
```diff
@@ -245,8 +245,8 @@ ThrowCompletionOr<String> temporal_year_month_to_string(GlobalObject& global_obj
     // 6. Let calendarID be ? ToString(yearMonth.[[Calendar]]).
     auto calendar_id = TRY(Value(&year_month.calendar()).to_string(global_object));
 
-    // 7. If calendarID is not "iso8601", then
-    if (calendar_id != "iso8601") {
+    // 7. If showCalendar is "always" or if calendarID is not "iso8601", then
+    if (show_calendar == "always" || calendar_id != "iso8601") {
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp:249]: ditto

File: Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp [hunk 3]
```diff
@@ -900,10 +926,25 @@ bool ISO8601Parser::parse_time_zone_iana_name_tail()
 bool ISO8601Parser::parse_time_zone_iana_name()
 {
     // TimeZoneIANAName :
-    //     TimeZoneIANANameTail
+    //     Etc/GMT ASCIISign UnpaddedHour
+    //     TimeZoneIANANameTail but not Etc/GMT ASCIISign UnpaddedHour
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:930]: That's a bit weird :thinking: what's the point of excluding a subset of a production and then immediately allowing it again

**Review by IdanHo — APPROVED** [review]
LGTM, just a couple of micro optimisations :^)

File: Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp (hunk 3 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:930]: Yeah, no idea :man_shrugging:

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainMonthDay.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/PlainMonthDay.cpp]: Done :+1:

File: Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp (hunk 3 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp:930]: Perhaps the author thought sub-productions need to mutually exclusive, even though ones that are not already exist...

**Pull request merged**